Fog.js

```diff
@@ -3196,7 +3215,7 @@ function redraw_light(){
   		drawPolygon(offscreenContext, lightPolygon, 'rgba(255, 255, 255, 1)', true); //draw to offscreen canvas so we don't have to render every draw and use this for a mask
 	}    	
   }
-	context.drawImage(offscreenCanvasMask, 0, 0); // draw to visible canvas only once so we render this once
+	await scheduler.postTask(() => context.drawImage(offscreenCanvasMask, 0, 0), {priority: "user-visible"}); // draw to visible canvas only once so we render this once

```

Comment:
Just was looking at this again, I _think_ if you move the above draw of a full black square used to reset the canvas to be in the same task here it will fix the flashing. I think what's happening is the black rect is being drawn the clear then this is taking some time to run?

Reply:
Ah yeah! That definitely would be it. Nice catch, thank you!

Reply:
That priority basically says prioritize user input over this task and will only run over tasks the browser deems as "background"

Reply:
ah that makes sense - thanks for explaining some of these comments are just my own lack of knowledge and picking your brain lol

Reply:
You give great feedback, so I always appreciate your reviews